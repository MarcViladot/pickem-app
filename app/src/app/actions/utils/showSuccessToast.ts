import i18next from 'i18next';
import Toast from 'react-native-toast-message';

export const SHOW_SUCCESS_TOAST = 'SHOW_SUCCESS_TOAST';
export const showSuccessToast = (msg?: string) => {
  Toast.show({
    text1: msg || i18next.t('COMMON.OPERATION_OK'),
    type: 'success',
    position: 'bottom',
  });
  return {
    type: SHOW_SUCCESS_TOAST,
  };
};
