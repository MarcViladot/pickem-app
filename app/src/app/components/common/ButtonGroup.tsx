import React, {FC, useEffect, useMemo, useState} from "react";
import {Pressable, StyleSheet, Text, TouchableOpacity, View} from "react-native";
import {useTranslation} from 'react-i18next';
import {Theme, useTheme} from '@react-navigation/native';

export interface IGroupButton {
    value: string | number | boolean;
    viewValue?: string;
}

interface IButton {
    button: IGroupButton;
    style: string;
}

interface Props {
    arrayOptions: IGroupButton[];
    initialValue: any;
    onSelect: (value: any) => void;
    fullWidth?: boolean;
}

const ButtonGroup: FC<Props> = ({arrayOptions, initialValue, onSelect, fullWidth}) => {

    const theme = useTheme();
    const {t} = useTranslation();
    const [selectedValue, setSelectedValue] = useState(initialValue);
    const [buttonsList, setButtonsList] = useState<IButton[]>([]);
    const styles = useMemo(() => createStyles(theme), [theme]);

    useEffect(() => {
        const buttons: IButton[] = arrayOptions.map((opt, i) => {
            return {
                button: opt,
                style: i === 0 ? 'firstButton' : i === arrayOptions.length - 1 ? 'lastButton' : ''
            }
        });
        setButtonsList(buttons);
    }, []);

    const selectValue = (value) => {
        if (value !== selectedValue) {
            setSelectedValue(value);
            onSelect(value);
        }
    };

    return (
        <View style={styles.mainView}>
            <View style={[styles.buttonGroup, {width: fullWidth ? '100%' : '82%'}]}>
                {buttonsList.map((option, i) => (
                        <Pressable onPress={() => selectValue(option.button.value)} key={i}
                                   testID={`Button${option.button.value}`}
                                   style={[styles.button, styles[option.style], option.button.value === selectedValue ? styles.selectedButton : styles.blankButton]}>
                            <Text
                                style={option.button.value === selectedValue ? styles.selectedButtonText : styles.blankButtonText}>
                                {option.button.viewValue ? t(option.button.viewValue) : option.button.value}
                            </Text>
                        </Pressable>
                    )
                )}
            </View>
        </View>
    );
};

const createStyles = (theme: Theme) =>
    StyleSheet.create({
        mainView: {
            flexDirection: 'row',
            justifyContent: 'center'
        },
        buttonGroup: {
            display: 'flex',
            flexDirection: 'row',
            height: 40,
            backgroundColor: theme.dark ? '#495666' : '#EBECF1',
            padding: 1,
            borderRadius: 50,
        },
        button: {
            borderRadius: 50,
            alignItems: 'center',
            flex: 1,
            justifyContent: 'center'
        },
        selectedButton: {
            backgroundColor: theme.dark ? '#202D3A' : '#FFF'
        },
        blankButton: {
            backgroundColor: theme.dark ? '#495666' : '#EBECF1'
        },
        selectedButtonText: {
            fontSize: 13,
            color: theme.colors.text
        },
        blankButtonText: {
            fontSize: 13,
            color: "#9DA3AC"
        },
        disabledButton: {
            backgroundColor: '#CCCCCC',
        }
    });

export default ButtonGroup;



