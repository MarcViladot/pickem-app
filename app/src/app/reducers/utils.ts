// import {SHOW_LOADING} from '../actions/utils/showLoading';
// import {HIDE_LOADING} from '../actions/utils/hideLoading';
// import {SET_LANG} from '../actions/utils/setLang';
import { AnyAction } from "redux";

const initialState = {
    serverLoading: false,
    currentLang: null
};

export default (state = initialState, action: AnyAction) => {
    /*if (action.type === SHOW_LOADING) {
        return {
            ...state,
            serverLoading: true
        };
    } else if (action.type === HIDE_LOADING) {
        return {
            ...state,
            serverLoading: false
        };
    } else if (action.type === SET_LANG) {
        return {
            ...state,
            currentLang: action.lang
        };
    }*/
    return state;
};
