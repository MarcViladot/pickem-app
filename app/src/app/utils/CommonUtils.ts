import {Round} from '../interfaces/league.interface';

export class CommonUtils {

    public static getRoundName(round: Round, lang: string): string {
        const translatedName = round.translationGroup.roundNames.find((trans) => trans.lang === lang);
        if (translatedName) {
            return `${translatedName.text} ${round.translationNameExtra}`;
        }
        return round.name;
    }
}
