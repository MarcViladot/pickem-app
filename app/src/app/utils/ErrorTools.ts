import {ResponseApi, WebApiCodesTranslationMap, WebApiErrorCode} from './IResponse';


export class ErrorTools {

    public static getApiErrorMsg(errorCode: number, params: string[]) {
        const error = WebApiCodesTranslationMap.get(errorCode) || 'ERRORS.UNEXPECTED_ERROR';
        const paramsToTranslate = {};
        if (params) {
            params.forEach((param, i) => paramsToTranslate[`p${i + 1}`] = param);
        }
        return {error, paramsToTranslate};
    }

    static getApiErrorMsgFromResponse(res: ResponseApi<any>) {
        return ErrorTools.getApiErrorMsg(res.ErrorCode, res.ErrorParams);
    }

}
