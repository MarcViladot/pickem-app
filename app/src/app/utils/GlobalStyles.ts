import { StyleSheet } from 'react-native';

export default StyleSheet.create({
    viewContainer: {
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
    },
    pageContainer: {
        padding: 5,
        flex: 1
    },
    pageContent: {
        width: '100%',
        flex: 1,
        flexDirection: 'column',
        backgroundColor: '#FFF',
        padding: 5,
        paddingTop: 0,
    },
    boldText: {
        fontWeight: 'bold'
    },
    row: {
        display: 'flex',
        flexDirection: 'row'
    },
    pageHeader: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        borderBottomWidth: 1,
        paddingBottom: 1,
        borderBottomColor: '#808080',
    },
    headerTitle: {
        fontSize: 22,
        fontFamily: 'Inter-Light'
    },
    dangerText: {
        color: '#BC1C61',
        textAlign: 'center',
        fontSize: 20,
    },
});
