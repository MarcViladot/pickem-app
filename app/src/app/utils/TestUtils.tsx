import store from '../../../store';
import {render} from '@testing-library/react-native'
import {Provider} from 'react-redux';
import React, {useState} from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {DarkTheme, LightTheme, ThemeContext} from '../../../themes';
import {ThemeProvider} from 'styled-components/native';

const AllTheProviders = ({children}) => {
    const [theme, setTheme] = useState('dark');
    const themeData = {theme, setTheme};
    return (
        <Provider store={store}>
            <ThemeContext.Provider value={themeData}>
                <ThemeProvider theme={theme === 'dark' ? DarkTheme : LightTheme}>
                    {children}
                </ThemeProvider>
            </ThemeContext.Provider>
        </Provider>
    )
};

const ReduxProvider = ({children}) => {

    return (
        <Provider store={store}>
            {children}
        </Provider>
    )
};

const ProvidersWithNav = ({children}) => {
    const [theme, setTheme] = useState('dark');
    const themeData = {theme, setTheme};
    return (
        <Provider store={store}>
            <ThemeContext.Provider value={themeData}>
                <ThemeProvider theme={DarkTheme}>
                    <NavigationContainer theme={DarkTheme}>
                        {children}
                    </NavigationContainer>
                </ThemeProvider>
            </ThemeContext.Provider>
        </Provider>
    )
};

const customRender = (ui, options?) =>
    render(ui, {wrapper: AllTheProviders, ...options});

const reduxRender = (ui, options?) =>
    render(ui, {wrapper: ReduxProvider, ...options});

const navRender = (ui, options?) =>
    render(ui, {wrapper: ProvidersWithNav, ...options});

export * from '@testing-library/react-native'
export {customRender as render}
export {reduxRender as reduxRender}
export {navRender as navRender}
