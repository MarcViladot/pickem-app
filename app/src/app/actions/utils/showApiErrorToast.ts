import {ResponseApiError, WebApiErrorCode} from '../../utils/IResponse';
import {ErrorTools} from '../../utils/ErrorTools';
import Toast from 'react-native-toast-message';
import i18next from 'i18next';

export const SHOW_TOAST = "SHOW_TOAST";
export const showApiErrorToast = (res: ResponseApiError) => {
    const translation = ErrorTools.getApiErrorMsgFromResponse(res);
    Toast.show({
        text1: i18next.t(translation.error, translation.paramsToTranslate),
        type: res.ErrorCode !== -2 ? 'apiError' : 'serverError',
        position: 'bottom',
        props: res
    });
    return {
        type: SHOW_TOAST
    }
};
