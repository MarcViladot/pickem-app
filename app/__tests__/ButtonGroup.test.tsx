import React from 'react';
import {fireEvent, waitFor} from '@testing-library/react-native';
import {render} from '../src/app/utils/TestUtils';
import ButtonGroup, {IGroupButton} from '../src/app/components/common/ButtonGroup';

describe('Button group component', () => {

    const mockOptions: IGroupButton[] = [
        {value: 1, viewValue: 'FIRST_VALUE'},
        {value: 2, viewValue: 'SECOND_VALUE'}
    ];
    const onSelected = jest.fn();

    it('renders correctly', async () => {
        const {toJSON} = await waitFor(() =>
            render(<ButtonGroup arrayOptions={mockOptions} onSelect={onSelected} initialValue={mockOptions[0].value} />)
        );
        expect(toJSON()).toMatchSnapshot();
    });

    it('should not call onSelect after button press if value does not change', async () => {
        const {getByTestId} = await waitFor(() =>
            render(<ButtonGroup arrayOptions={mockOptions} onSelect={onSelected} initialValue={mockOptions[0].value} />)
        );
        const sameValue = mockOptions[0].value;
        const button = getByTestId(`Button${sameValue}`);
        fireEvent.press(button);
        expect(onSelected).not.toHaveBeenCalled();
    });

    it('should call onSelect after button press', async () => {
        const {getByTestId} = await waitFor(() =>
            render(<ButtonGroup arrayOptions={mockOptions} onSelect={onSelected} initialValue={mockOptions[0].value} />)
        );
        const newValue = mockOptions[1].value;
        const button = getByTestId(`Button${newValue}`);
        fireEvent.press(button);
        expect(onSelected).toHaveBeenCalledWith(newValue);
    });

});
