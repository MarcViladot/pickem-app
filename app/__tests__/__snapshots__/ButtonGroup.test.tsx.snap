// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button group component renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#495666",
        "borderRadius": 50,
        "display": "flex",
        "flexDirection": "row",
        "height": 40,
        "padding": 1,
        "width": "82%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
          Object {
            "backgroundColor": "#202D3A",
          },
        ]
      }
      testID="Button1"
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 13,
          }
        }
      >
        FIRST_VALUE
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
          Object {
            "backgroundColor": "#495666",
          },
        ]
      }
      testID="Button2"
    >
      <Text
        style={
          Object {
            "color": "#9DA3AC",
            "fontSize": 13,
          }
        }
      >
        SECOND_VALUE
      </Text>
    </View>
  </View>
</View>
`;
